src/olympia/internal_tools/views.py

```diff
@@ -29,17 +35,32 @@ class InternalAddonSearchView(AddonSearchView):
 class LoginStart(APIView):
 
     def get(self, request):
+        config = settings.FXA_CONFIG['internal']
         request.session.setdefault('fxa_state', generate_fxa_state())
         state = request.session['fxa_state']
         next_path = request.GET.get('to')
         if next_path and is_safe_url(next_path):
             state += ':' + urlsafe_b64encode(next_path).rstrip('=')
         query = {
-            'client_id': settings.ADMIN_FXA_CONFIG['client_id'],
-            'redirect_uri': settings.ADMIN_FXA_CONFIG['redirect_uri'],
-            'scope': settings.ADMIN_FXA_CONFIG['scope'],
+            'client_id': config['client_id'],
+            'redirect_uri': config['redirect_uri'],
+            'scope': config['scope'],
             'state': state,
         }
         return HttpResponseRedirect('{host}/authorization?{query}'.format(
-            host=settings.ADMIN_FXA_CONFIG['oauth_host'],
+            host=config['oauth_host'],
             query=urlencode(query)))
+
+
+class LoginView(APIView):
+
+    @with_user(format='json', config='internal')
+    def post(self, request, user, identity, next_path):
+        if user is None:
+            return Response({'error': ERROR_NO_USER}, status=422)
```

Comment:
Can we use `rest_framework.status` shortcuts to make the status codes more readable? I don't think we do that anywhere currently but I just noticed.